### PR TITLE
Added categorization

### PR DESCRIPTION
**Pull request by JulianKnodt**

### CODE REVIEW

File: router/clientRouter.js [hunk 1]
```diff
@@ -24,8 +25,12 @@ router
 
 // Images Routes
   .post('/image', images.addImage)
-  .get('recent', suggestions.getRecent)
-  .get('suggestions', suggestions.getSuggestions);
+  .get('/api/recent', suggestions.getRecent)
```
Comment (drawnwren) [inline router/clientRouter.js]: Do you intend for these routes to be /api/api/recent? The client router is used inside of the /api URI

**Comment by drawnwren** [comment]
I'm not sure which branch (client or server) is causing this, but when I try to add an Item with categories I get a db error: Creating new item, { title: 'Optimus Change Bot',
  description: 'Not mason. Would like your change before it changes you!',
  price: '1',
  category: '[]',
  created_at: '2016-10-12T02:39:27.698Z',
  updated_at: '2016-10-12T02:39:27.698Z',
  images: '["https://bitbargainbucket.s3.amazonaws.com/images/test2/changeintoatruck.jpg"]' }
Inserting,  { title: 'Optimus Change Bot',
  description: 'Not mason. Would like your change before it changes you!',
  price: '1',
  category: '[]',
  created_at: '2016-10-12T02:39:27.698Z',
  updated_at: '2016-10-12T02:39:27.698Z',
  images: '["https://bitbargainbucket.s3.amazonaws.com/images/test2/changeintoatruck.jpg"]' } object
POST /api/items/sell 200 183.825 ms - 340
Error inserting into transactions table error: insert into "transactions" ("buyer_id", "item_id", "seller_id") values ($1, $2, $3) returning "id" - insert or update on table "transactions" violates foreign key constraint "transactions_seller_id_foreign"

Also of note, I entered 3 categories for that item which didn't make it to the server. Robots, Presidential, Chickens for America, Not Mason

**Comment by drawnwren** [comment]
It might be best to include some tests with this to show that it's working. (We are going to be writing them over the next few days anyways.)

**Review by chrononaut — APPROVED** [review]

File: controllers/categoryController.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    //Send data to categorization service's /predict route
+    request.post({url: 'http://localhost:5000/predict', form:{data: req.body.text} }, (err, resp, body) => {
+      res.send(body);
+    });
+  },
+  //Fits the model of the service to fit user input(This route is currently never called but should work)
+  //Does not return anything except 200 status
+  fit: (req, res) => {
+    request.post({url: 'http://localhost:5000/fit', form:{categories:req.body.categories} }, (err, resp, body) => {
+      console.log(err, resp, body);
```
Comment (RSiferd2) [inline controllers/categoryController.js]: Could you remove this console log

File: controllers/transactionsController.js [hunk 3]
```diff
@@ -37,18 +39,20 @@ module.exports = {
...
     .then(users => {
+      console.log(users);
       buyer = users[0].email;
       seller = users[1].email;
     })
     .then(() => bitcoin.move(buyer, seller, remainder))
     .then(result => bitcoin.move(buyer, fees, fee))
     .then(result => res.send('Payment sent'))
-    .catch(err => console.error('Error sending payment'), err);
+    .catch(err => console.error('Error sending payment', err));
```
Comment (RSiferd2) [inline controllers/transactionsController.js]: I thought we were moving all console logs to next calls for Morgan to handle?

File: controllers/transactionsController.js [hunk 4]
```diff
@@ -37,18 +39,20 @@ module.exports = {
...
   sendPayment(req, res) {
+    const amount = Number(req.body.price);
     const fee = amount * 0.01, remainder = amount - fee;
     let buyer = '', seller = '';
 
     db.transactions.getByItemId(req.body.id)
-    .then(item => Promise.all([db.users.getById(item[seller_id]), db.users.getById(item[buyer_id])])
+    .then(item => Promise.all([db.users.getById(item['seller_id']), db.users.getById(item['buyer_id'])]))
     .then(users => {
+      console.log(users);
```
Comment (RSiferd2) [inline controllers/transactionsController.js]: Remove please

File: search/elasticSearch.js [hunk 5]
```diff
@@ -15,7 +15,7 @@ class ElSearch {
    */
   constructor() {
     this.client = new elasticsearch.Client({
-      host: 'localhost:9200'
+      host: '52.52.22.4:9200'
```
Comment (RSiferd2) [inline search/elasticSearch.js]: Maybe we should remove all hostnames and ports to one of the config files and export them to the process env?

**Review by RSiferd2 — CHANGES_REQUESTED** [review]

**Review by RSiferd2 — APPROVED** [review]

**Pull request merged**